h5py/h5t.pyx

```diff
@@ -218,6 +219,11 @@ PYTHON_OBJECT = lockid(H5PY_OBJ)
 cdef dict _order_map = { H5T_ORDER_NONE: '|', H5T_ORDER_LE: '<', H5T_ORDER_BE: '>'}
 cdef dict _sign_map  = { H5T_SGN_NONE: 'u', H5T_SGN_2: 'i' }
 
+# Available floating point types
+available_ftypes = dict()
+for ftype in np.typeDict.values():
+    if np.issubdtype(ftype, float):
+        available_ftypes[np.dtype(ftype).itemsize] = np.finfo(ftype)
```

Comment:
I can't see this being a problem, so long as numpy never supports creating new types which get linked into the standard types.